### PR TITLE
PV2-1208 Added aim sequence number to the EarningEvent table

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/SFA.DAS.Payments.EarningEvents.Application/Mapping/EarningsEventProfile.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ public EarningsEventProfile()
                 .ForMember(dest => dest.ProgrammeType, opt => opt.MapFrom(source => source.Aim.LearningDeliveryValues.ProgType))
                 .ForMember(dest => dest.Reference, opt => opt.MapFrom(source => source.Aim.LearningDeliveryValues.LearnAimRef))
                 .ForMember(dest => dest.StandardCode, opt => opt.MapFrom(source => source.Aim.LearningDeliveryValues.StdCode))
+                .ForMember(dest => dest.SequenceNumber, opt => opt.MapFrom(source => source.Aim.AimSeqNumber))
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.EarningEvents.Application/Mapping/EarningsEventProfile.cs]: I can't see any Mapping Unit test that shows that this was mapped correctly.

Comment (carlbm) [reply src/SFA.DAS.Payments.EarningEvents.Application/Mapping/EarningsEventProfile.cs]: Test added

File: src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.EarningEvent.sql [hunk 2]
```diff
@@ -20,5 +20,5 @@
 	JobId  BIGINT NOT NULL,
 	EventTime DATETIMEOFFSET NOT NULL,
 	CreationDate DATETIMEOFFSET NOT NULL CONSTRAINT DF_EarningEvent__CreationDate DEFAULT (SYSDATETIMEOFFSET()), 
-    [LearningAimSequenceNumber] BIGINT NULL
+    [LearningAimSequenceNumber] BIGINT NOT NULL
```
Comment (abroodie) [inline src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.EarningEvent.sql]: You will either need to leave this as null or add a pre/post deploy script to manage this change as the app is now in several environments where clearing all tables is not a reasonable option.

File: src/SFA.DAS.Payments.Audit.Model/EarningEventModel.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@ public class EarningEventModel: PaymentsEventModel
     {
         public ContractType ContractType { get; set; }
         public string AgreementId { get; set; }
+        public int LearningAimSequenceNumber { get; set; }
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/SFA.DAS.Payments.Audit.Model/EarningEventModel.cs]: Updated

File: src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.EarningEvent.sql (hunk 2 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Database/Payments2/Tables/Payments2.EarningEvent.sql]: Updated

**Pull request merged**